proposals/cleanup.md

```diff
@@ -233,6 +233,87 @@ Kyverno could alternatively leverage a CronJob resource to perform the deletions
+
+
+
+**Flow:**
+
+1. A new policy is added to the cluster. 
+2. The cleanup policy object is added to the controller queue with the help of CleanupPolicy and ClusterCleanupPolicy informers.
+3. The controller then gets the policy.
+4. Get the matching resources for that rule/policy.
+5. Create CronJob for scheduling the deletion for each resource.
+6. Kyverno’s Service account is used in CronJob.
+7. OwnerReference of the CronJob is set to the trigger resource, so when the matching resource is deleted, the CJ is also deleted.
+
+**Life Cycle of the CronJob:**
+
+In this approach lifecycle of the CronJob created will be same as the matching resource. Since the OwnerReference of the matching resource will be set to the mathcing and this way CronJob's garbase collection will be handled.
+
+| Pros | Cons |
+| --- | --- |
+| Deletion of CJs after resource cleanup can be easily handled through OwnerReference | Creating CJs for each resource could be inefficient in large clusters. |
+
+
+### 2. Creating CronJobs per rule/policy.
+A cleanup controller will be creating CronJob per rule/policy. Creation and updation of a new cleanup policy will trigger the reconciliation. When reconciliation is triggered, a CronJob is created and delete commands for all the matching resources are informed to the CronJob and are executed at the same schedule (specified in the policy).
+
+
+
+**Flow:**
+
+1. A new policy is added to the cluster. 
+2. The policy object is added to the controller queue  with the help of CleanupPolicy and ClusterCleanupPolicy informers.
+3. Get the matching resources for that policy. The matching resources could be of different kinds.
+4. Delete commands for all the matching resources are informed to the CronJob and are executed at the same schedule.
+5. On getting the admission request, fetch the CronJob created for the policy with the help of labels consisting of the rule and policy name.
+6. Kyverno’s Service account is used in CronJob.
+
+In this approach, CronJob is going to need to run the reconciliation logic for processing the rule:
+
+1. CronJob runs a processing logic for applying the rule to the resources evaluate the resource against the engine to determine if the resource applies. 
+2. If the resource applies, delete the resource.
+
+This logic exists in Kyverno, so CronJobs will be created based on engine responses. Containerize this logic and build an image for it to run in the CronJob.
+
+**Life Cycle of the CronJob:**
+
+In this approach the life-cycle of the CronJob will same as the CleanupPolicy or ClusterCleanupPolicy resource which triggers the creation of the CronJob
```

Comment:
In this approach the life-cycle of the CronJob will same as the CleanupPolicy or **the** ClusterCleanupPolicy ~~resource which triggers the creation of the CronJob~~.